base/src/main/java/com/smartdevicelink/managers/screen/SoftButtonObject.java

```diff
@@ -264,6 +275,30 @@ public void setName(@NonNull String name) {
      * @param states a list of the object's soft button states. <strong>states should be unique for every SoftButtonObject. A SoftButtonState instance cannot be reused for multiple SoftButtonObjects.</strong>
      */
     public void setStates(@NonNull List<SoftButtonState> states) {
+
+        boolean repeatedStateNames = hasTwoStatesOfSameName(states);
+
+        if (repeatedStateNames) {
+            DebugTool.logError(TAG, "A SoftButtonObject must have states with different names.");
+            return;
+        }
+
+        if (states.isEmpty()) {
+            DebugTool.logError(TAG, "A SoftButtonObject must contain at least one state");
+            return;
+        }
+
+        boolean hasStateWithCurrentName = false;
+        for (SoftButtonState state : states) {
+            if(state.getName().equals(currentStateName)) {
+                hasStateWithCurrentName = true;
+                break;
+            }
+        }
+        if (!hasStateWithCurrentName) {
+            DebugTool.logError(TAG, "A SoftButtonObject should have a state with currentStateName.");
```

Comment:
```suggestion
            DebugTool.logError(TAG, "A SoftButtonObject setting states must contain a state with currentStateName.");
```

Can you also include the name of `currentStateName`?